internal/webhook/notebook/mutating.go

```diff
@@ -0,0 +1,310 @@
+		secretParts := strings.Split(part, "/")
+		if len(secretParts) != 2 {
+			return nil, fmt.Errorf("invalid secret reference format '%s' - expected 'namespace/name'", part)
+		}
+
+		namespace := strings.TrimSpace(secretParts[0])
+		name := strings.TrimSpace(secretParts[1])
+
+		if namespace == "" || name == "" {
+			return nil, fmt.Errorf("invalid secret reference '%s' - namespace and name cannot be empty", part)
+		}
+
+		secretRefs = append(secretRefs, corev1.SecretReference{
+			Name:      name,
+			Namespace: namespace,
+		})
+	}
+
+	return secretRefs, nil
+}
+
+// checkSecretPermission checks if the user has permission to "get" the specified secret using SubjectAccessReview.
+func (w *NotebookWebhook) checkSecretPermission(ctx context.Context, req *admission.Request, secretRef corev1.SecretReference) (bool, error) {
+	log := logf.FromContext(ctx)
+
+	// Create a SubjectAccessReview to check if the user can "get" the secret
+	sar := &authorizationv1.SubjectAccessReview{
+		Spec: authorizationv1.SubjectAccessReviewSpec{
+			User:   req.UserInfo.Username,
+			Groups: req.UserInfo.Groups,
+			ResourceAttributes: &authorizationv1.ResourceAttributes{
+				Namespace: secretRef.Namespace,
+				Verb:      "get",
+				Group:     "",
+				Version:   "v1",
+				Resource:  "secrets",
+				Name:      secretRef.Name,
+			},
+		},
+	}
+
+	// Send the SubjectAccessReview to the API server
+	if err := w.Client.Create(ctx, sar); err != nil {
+		log.Error(err, "failed to create SubjectAccessReview", "secret", secretRef.Name, "namespace", secretRef.Namespace)
+		return false, fmt.Errorf("failed to create SubjectAccessReview: %w", err)
+	}
+
+	// Check the result
+	if !sar.Status.Allowed {
+		log.V(1).Info("user does not have permission to access secret",
+			"secret", secretRef.Name,
+			"namespace", secretRef.Namespace,
+			"reason", sar.Status.Reason,
+			"evaluationError", sar.Status.EvaluationError,
+		)
+
+		return false, nil
+	}
+
+	log.V(1).Info("user has permission to access secret", "secret", secretRef.Name, "namespace", secretRef.Namespace)
+	return true, nil
+}
+
+func (w *NotebookWebhook) performConnectionInjection(req admission.Request, secretRefs []corev1.SecretReference) (bool, *unstructured.Unstructured, error) {
+	obj := &unstructured.Unstructured{}
+	if err := w.Decoder.Decode(req, obj); err != nil {
+		return false, nil, fmt.Errorf("failed to decode Notebook object: %w", err)
+	}
+
+	if len(secretRefs) == 0 {
+		return false, obj, nil
+	}
+
+	// Get the notebook containers
+	containers, found, err := unstructured.NestedSlice(obj.Object, NotebookContainersPath...)
+	if err != nil {
+		return false, nil, fmt.Errorf("failed to get containers array: %w", err)
+	}
+	if !found || len(containers) == 0 {
+		return false, nil, errors.New("no containers found in notebook")
+	}
+
+	// The notebook only has one container, so we can get the envFrom from the first container
+	container, ok := containers[0].(map[string]interface{})
```

Comment:
i am thinking about this part.
so the injection of secret is only done to the first one?
e.g i have a main container and another oauth-proxy container, it might be in the main or also could be in oauth-proxy?

Reply:
You're right in gitopt this isn't always necessarily true. But I'm not sure how to determine what the right container is in this scenario because the user could name the container anything. Do we want to document and enforce that the name of the container be a specific, known value?

Reply:
what if , a simple way, just inject to all containers?

Reply:
It's fine for now to mimic the dashboard behavior because we already know that works, doing so leaves one less difference to investigate if we run into bugs later, and we know that no gitops users make use of the existing connection binding machinery because it's tucked away in dashboard code.